python/setup.py

```diff
@@ -35,59 +42,143 @@
     cuda_include_dir = os.path.join(os.environ.get('CUDA_HOME'), 'include')
 
 
-rmm_include_dir = '/include'
-rmm_lib_dir = '/lib'
-
-# including cumlprims dirs, in case rmm paths change in the future
-cumlprims_include_dir = '/include'
-cumlprims_lib_dir = '/lib'
+conda_lib_dir = os.path.normpath(sys.prefix) + '/lib'
+conda_include_dir = os.path.normpath(sys.prefix) + '/include'
 
 if os.environ.get('CONDA_PREFIX', None):
     conda_prefix = os.environ.get('CONDA_PREFIX')
-    rmm_include_dir = conda_prefix + rmm_include_dir
-    rmm_lib_dir = conda_prefix + rmm_lib_dir
+    conda_include_dir = conda_prefix + '/include'
+    conda_lib_dir = conda_prefix + '/lib'
+
+
+##############################################################################
+# - Subrepo checking and cloning ---------------------------------------------
+
+
+def clone_repo(name, GIT_REPOSITORY, GIT_TAG, first=False):
+    """
+    Function to clone repos in case they were not cloned by cmake.
+    Variables are named identical to the cmake counterparts for clarity,
+    in spite of not being very pythonic.
+    """
+    if first:
+        subprocess.check_call(['rm', '-rf', 'external'])
+
+    subprocess.check_call(['git', 'clone',
+                           GIT_REPOSITORY,
+                           'external/' + name])
+    wd = os.getcwd()
+    os.chdir("external/" + name)
+    subprocess.check_call(['git', 'checkout',
+                          GIT_TAG])
+    os.chdir(wd)
+
+
+if "clean" not in sys.argv:
+    if os.environ.get('CUML_BUILD_PATH', False):
+        libcuml_path = '../' + os.environ.get('CUML_BUILD_PATH')
+    else:
+        libcuml_path = '../cpp/build/'
+
+    # Cloning repos if they are not found in libcuml_path (derived from
+    # CUML_BUILD_PATH) This should match their equivalent repos and tags of the
+    # CMakeLists that was used to build libcuml++
+    if not os.path.exists(libcuml_path):
+        warnings.warn("Third party repositories have not been found so they  "
+                      "will be cloned. To avoid this set the environment "
+                      "variable CUML_BUILD_PATH, containing the relative "
+                      "path of the root of the repository to the folder "
+                      "where libcuml++ was built.")
+
+        clone_repo(name='cub',
+                   GIT_REPOSITORY='https://github.com/NVlabs/cub.git',
+                   GIT_TAG='v1.8.0',
+                   first=True)
+        cub_path = 'external/cub'
+
+        clone_repo(name='cutlass',
+                   GIT_REPOSITORY='https://github.com/NVIDIA/cutlass.git',
+                   GIT_TAG='v1.0.1')
+        cutlass_path = 'external/cutlass'
+
+        clone_repo(name='faiss',
+                   GIT_REPOSITORY='https://github.com/facebookresearch/faiss.git',  # noqa: E501
+                   GIT_TAG='656368b5eda4d376177a3355673d217fa95000b6')
+        # Note: faiss has a different include path
+        faiss_path = 'external'
```

Comment:
maybe `external/faiss`?

Reply:
No, given the faiss non standard structure, it has to be like this. These paths depend on the sub projects folder structure